### PR TITLE
Update outdated content

### PR DESCRIPTION
**Pull request by haydentherapper**
This includes updates to the FAQs, security model, and more details about signing and verification.

Updates:

* FAQ - Removed outdated information, added details about new clients and log monitor
* How Sigstore works ecosystem page (groups) - Proof read, clarified some points
* How it works - Updated outdated examples repo
* Software list - Cleaned up descriptions
* Trust and security - Updated threat model to security model, added link

### CODE REVIEW

**Review by bobcallaway — APPROVED** [review]

File: content/software/software_list.json [hunk 1]
```diff
@@ -26,19 +26,15 @@
...
-      "productOverview": "A free root certification authority, issuing temporary certificates to an authorized identity and publishing them in the Rekor transparency log.",
+      "productOverview": "Code-signing certificate authority, issuing short-lived certificates to an authenticated identity and publishing them to a certificate transparency log.",
       "productCTA": "View the repo",
       "productCTALink": "https://github.com/sigstore/fulcio"
     },
     {
       "productName": "Trust root",
-      "productOverview": "The foundation for trust behind the whole of sigstore utilizes TUF. This repository describes this process, our keyholders, and ways of working to protect the root keys.",
-      "productCTA": "sigstore's trust root",
+      "productOverview": "The foundation for trust underpinning Sigstore utilizes TUF. This repository describes this process, our keyholders, andhow the root keys are protected.",
```
Comment (ltagliaferri) [inline content/software/software_list.json:35]:
```suggestion
      "productOverview": "The foundation for trust underpinning Sigstore utilizes TUF. This repository describes this process, our keyholders, and how the root keys are protected.",
```

File: content/groups/groups.json [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
+      "textOverview": "First, tell sigstore to do the signing for you. Using Fulcio, sigstore requests a certificate from our Certificate Authority (CA). This checks you are who you say you are using OpenID Connect, which looks at an identity, such as an email address, to prove you’re the author. Fulcio grants a short-lived certificate, associating a provided public key with your identity.\n\nYou don’t have to manage keys yourself, and sigstore never obtains your private key. The public key that Cosign creates gets bound to your certificate.\n\nYour certificate then comes back to a sigstore client, which signs an artifact. The artifact hash, signature, and certificate are uploaded to a Rekor transparency log, for public auditability of the signing event."
```
Comment (ltagliaferri) [inline content/groups/groups.json:6]:
```suggestion
      "textOverview": "First, tell Sigstore to do the signing for you. Using Fulcio, Sigstore requests a certificate from our Certificate Authority (CA). This checks you are who you say you are using OpenID Connect, which looks at an identity, such as an email address, to prove you’re the author. Fulcio grants a short-lived certificate, associating a provided public key with your identity.\n\nYou don’t have to manage keys yourself, and Sigstore never obtains your private key. The public key that Cosign creates gets bound to your certificate.\n\nYour certificate then comes back to a Sigstore client, which signs an artifact. The artifact hash, signature, and certificate are uploaded to a Rekor transparency log, for public auditability of the signing event."
```

**Review by ltagliaferri — APPROVED** [review]
Thanks! Just a few tiny copyedits

File: content/groups/groups.json [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
+      "textOverview": "To sign your release, an Open ID Connect token exchange is first needed to generate a certificate. It’s automatic, one click authorization that says you are who you say you are. This certificate gets entered into the transparency log. If you’re not using Cosign’s keyless signing commands you’d still need to export your public key during this step.\n\n`rekor upload` then publishes your signature along with the artifact URL to the transparency log. Rekor verifies everything, downloads a local copy of your artifact and validates the artifact signing. The entry will be made into the log and you’ll get a URL.\n\nWhoever downloads your signed artifact can now look up the signature and URL to verify an inclusion proof, showing the unique information about that artifact proving where, when and who it came from."
```
[the remark this answers is not in the document]
Comment (haydentherapper) [reply content/groups/groups.json]: Done!

**Comment by haydentherapper** [comment]
Done! We'll want to make a pass over all of the content at some point for grammar and stylizing Sigstore.

**Comment by haydentherapper** [comment]
@ltagliaferri Should be ready to merge

**Comment by haydentherapper** [comment]
Bumping for a merge, want to get this in soon for any newcomers

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**